client.go

```diff
@@ -186,6 +186,7 @@ func NewWithConn(conn *grpc.ClientConn, opts ...ClientOpt) (*Client, error) {
 // Client is the client to interact with containerd and its various services
 // using a uniform interface
 type Client struct {
+	decryptionKeys []string
```

Comment:
I don't like this method at all.  The codebase should be unaware of encryption decryption except for the methods that need it.  This needs to be a simple, `Opt` that takes the inputs required to do enc/dec.

Reply:
Moved into options.